Wrappers/Python/cil/processors/FluxNormaliser.py

```diff
@@ -0,0 +1,314 @@
+#  Copyright 2024 United Kingdom Research and Innovation
+#  Copyright 2024 The University of Manchester
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# Authors:
+# CIL Developers, listed at: https://github.com/TomographicImaging/CIL/blob/master/NOTICE.txt
+
+from cil.framework import Processor, DataContainer, AcquisitionData,\
+ AcquisitionGeometry, ImageGeometry, ImageData
+from cil.utilities.display import show2D
+import numpy
+import logging
+import matplotlib.pyplot as plt
+
+log = logging.getLogger(__name__)
+
+class FluxNormaliser(Processor):
+    r'''
+    Flux normalisation based on float or region of interest
+
+    This processor reads in a AcquisitionData and normalises it based on
+    a float or array of float values, or a region of interest.
+    
+    The normalised image :math:`I_{norm}` is calculated from the original image
+    :math:`I` by
+
+    .. math:: I_{norm} = I\frac{n}{F}
+
+    where :math:`F` is the flux and :math:`n` is the norm_value
+
+    Parameters:
+    -----------
+    flux: float or list of floats (optional)
+        The value to divide the image by. If flux is a list it must have length 
+        equal to the number of angles in the dataset. If flux=None, calculate
+        flux from the roi, default is None.
+    
+    roi: dict (optional)
+        Dictionary describing the region of interest containing the background
+        in the image. The image is divided by the mean value in the roi. If None,
+        specify flux direcrtly, default is None.
+
+    norm_value: float (optional)
+        The value to multiply the image by. If None, the mean flux value will 
+        be used, default is None.
+        
+    tolerance: float (optional)
```

Comment:
is tolerance the right term?

Reply:
Hi both, I've removed this argument and instead raise an error if the flux contains a 0.